packages/framer-motion/src/animation/optimized-appear/start.ts

```diff
@@ -87,17 +109,39 @@ export function startOptimizedAppearAnimation(
             }
         }
 
-        /**
-         * Keep a map of elementIds that have started animating. We check
-         * via ID instead of Element because of hydration errors and
-         * pre-hydration checks. We also actively record IDs as they start
-         * animating rather than simply checking for data-appear-id as
-         * this attrbute might be present but not lead to an animation, for
-         * instance if the element's appear animation is on a different
-         * breakpoint.
-         */
-        window.MotionHasOptimisedAnimation = (elementId?: string) =>
-            Boolean(elementId && elementsWithAppearAnimations.has(elementId))
+        window.MotionCheckAppearSync = (
+            visualElement: WithAppearProps,
+            valueName: string,
+            value: MotionValue
+        ) => {
+            const appearId = getOptimisedAppearId(visualElement)
+
+            if (!appearId) return
+
+            const valueIsOptimised = window.MotionHasOptimisedAnimation?.(
+                appearId,
+                valueName
+            )
+            const externalAnimationValue =
+                visualElement.props.values?.[valueName]
+
+            if (!valueIsOptimised || !externalAnimationValue) return
+
+            const removeSyncCheck = value.on(
+                "change",
+                (latestValue: string | number) => {
+                    if (externalAnimationValue.get() !== latestValue) {
+                        window.MotionCancelOptimisedAnimation?.(
+                            appearId,
+                            valueName
+                        )
+                        removeSyncCheck()
```

Comment:
I'm not super familiar with the `value.on` API, why do we need to call `removeSyncCheck` again?

Reply:
Think of it like 

```
function doThing() {
  if (someCondition) {
    element.removeEventListener("mousestart", doThing)
  }
}

element.addEventListener("mousestart", doThing)
```